node/src/Pos/Wallet/Web/State/Transactions.hs

```diff
@@ -0,0 +1,104 @@
+createAccountWithAddress
+  :: AccountId
+  -> CAccountMeta
+  -> CWAddressMeta
+  -> Update ()
+createAccountWithAddress accId accMeta addrMeta = do
+  WS.createAccount accId accMeta
+  WS.addWAddress addrMeta
+
+-- | Delete a wallet (and all associated data).
+--   Compared to the low-level 'removeWallet', this function:
+--   - Removes all accounts associated with the wallet.
+--   - Removes transaction metadata.
+--   - Removes the history cache.
+deleteWallet
+  :: CId Wal
+  -> Update ()
+deleteWallet walId = do
+  accIds <- filter ((== walId) . aiWId) . HM.keys <$> use WS.wsAccountInfos
+  for_ accIds WS.removeAccount
+  WS.removeWallet walId
+  WS.removeTxMetas walId
+  WS.removeHistoryCache walId
+
+-- | Apply some set of modifiers to a wallet.
+--   TODO Find out the significance of this set of modifiers and document.
+applyModifierToWallet
+  :: CId Wal
+  -> [CWAddressMeta] -- ^ Wallet addresses to add
+  -> [(WS.CustomAddressType, [(CId Addr, HeaderHash)])] -- ^ Custom addresses to add
+  -> UtxoModifier
+  -> [(CTxId, CTxMeta)] -- ^ Transaction metadata to add
+  -> Map TxId TxHistoryEntry -- ^ Entries for the history cache
+  -> [(TxId, PtxCondition)] -- ^ PTX Conditions
+  -> HeaderHash -- ^ New sync tip
+  -> Update ()
+applyModifierToWallet walId wAddrs custAddrs utxoMod
+                      txMetas historyEntries ptxConditions
+                      syncTip = do
+  for_ wAddrs WS.addWAddress
+  for_ custAddrs $ \(cat, addrs) ->
+    for_ addrs $ WS.addCustomAddress cat
+  WS.updateWalletBalancesAndUtxo utxoMod
+  for_ txMetas $ uncurry $ WS.addOnlyNewTxMeta walId
+  WS.insertIntoHistoryCache walId historyEntries
+  for_ ptxConditions $ uncurry $ WS.setPtxCondition walId
+  WS.setWalletSyncTip walId syncTip
+
+-- | Rollback some set of modifiers to a wallet.
+--   TODO Find out the significance of this set of modifiers and document.
+rollbackModifierFromWallet
+  :: CId Wal
+  -> [CWAddressMeta] -- ^ Addresses to remove
+  -> [(WS.CustomAddressType, [(CId Addr, HeaderHash)])] -- ^ Custom addresses to remove
+  -> UtxoModifier
+     -- We use this odd representation because Data.Map does not get 'withoutKeys'
+     -- until 5.8.1
+  -> Map TxId () -- ^ Entries to remove from history cache.
+  -> [(TxId, PtxCondition, WS.PtxMetaUpdate)] -- ^ Deleted PTX candidates
+  -> HeaderHash -- ^ New sync tip
+  -> Update ()
+rollbackModifierFromWallet walId wAddrs custAddrs utxoMod
+                           historyEntries ptxConditions
+                           syncTip = do
+  for_ wAddrs WS.removeWAddress
+  for_ custAddrs $ \(cat, addrs) ->
+    for_ addrs $ WS.removeCustomAddress cat
+  WS.updateWalletBalancesAndUtxo utxoMod
+  WS.removeFromHistoryCache walId historyEntries
+  WS.removeWalletTxMetas walId (encodeCType <$> M.keys historyEntries)
+  for_ ptxConditions $ \(txId, cond, meta) -> do
+    WS.ptxUpdateMeta walId txId meta
+    WS.setPtxCondition walId txId cond
+  WS.setWalletSyncTip walId syncTip
```

Comment:
Please reformat this module to use 4-spaces indentation (as it's done in all other code in `cardano-sl`) 🙏

Reply:
Haha stylish-haskell police to the rescue! 😁  Don't worry @gromakovsky , the agreement with Nic is that once we are confident this PR delivers what it should, he will rebase everything properly (by adding, for example, the `[CSL-xxxx]` prefix to the commits) as well as fixing stylish issues 😁

Reply:
Yeah, I'm aware of the need to do this, but 4-space indentation makes me miserable so I'm putting it off :-P

Reply:
@gromakovsky Speaking of code style, do you have the appropriate `stylish-haskell.yaml` file for master?

Reply:
See [this thread](https://input-output-rnd.slack.com/archives/C2VJ41WDP/p1515686971000339) about `stylish-haskell` and `master`.